File: VB/PayrollBusiness.Setting.vb
Imports PayrollBusiness.ServiceContracts
Imports PayrollDAL
Imports Framework.Data

Namespace PayrollBusiness.ServiceImplementations
    Partial Public Class PayrollBusiness
        Implements ServiceContracts.IPayrollBusiness

#Region "Hold Salary"

        Public Function GetHoldSalaryList(ByVal PeriodId As Integer, ByVal OrgId As Integer, ByVal IsDissolve As Integer, ByVal log As UserLog,
                                        ByVal PageIndex As Integer,
                                        ByVal PageSize As Integer,
                                        ByRef Total As Integer,
                                        Optional ByVal Sorts As String = "CREATED_DATE desc") As List(Of PAHoldSalaryDTO) Implements ServiceContracts.IPayrollBusiness.GetHoldSalaryList
            Try
                Dim rep As New PayrollRepository
                Return rep.GetHoldSalaryList(PeriodId, OrgId, IsDissolve, log, PageIndex, PageSize, Total, Sorts)
            Catch ex As Exception

                Throw ex
            End Try
        End Function

        Public Function InsertHoldSalary(ByVal objPeriod As List(Of PAHoldSalaryDTO), ByVal log As UserLog) As Boolean Implements ServiceContracts.IPayrollBusiness.InsertHoldSalary
            Try
                Dim rep As New PayrollRepository
                Return rep.InsertHoldSalary(objPeriod, log)
            Catch ex As Exception

                Throw ex
            End Try
        End Function

        Public Function DeleteHoldSalary(ByVal lstDelete As List(Of Decimal)) As Boolean Implements ServiceContracts.IPayrollBusiness.DeleteHoldSalary
            Try
                Dim rep As New PayrollRepository
                Return rep.DeleteHoldSalary(lstDelete)
            Catch ex As Exception
                Throw ex
            End Try
        End Function

#End Region


    End Class
End Namespace

File: C#/PayrollBusiness.Setting.cs
using System;
using PayrollBusiness.ServiceContracts;
using PayrollDAL;
using Framework.Data;

namespace PayrollBusiness.ServiceImplementations
{
    public partial class PayrollBusiness : ServiceContracts.IPayrollBusiness
    {
        #region "Hold Salary"
        public List<PAHoldSalaryDTO> GetHoldSalaryList(int PeriodId, int OrgId, int IsDissolve, UserLog log, int PageIndex, int PageSize, ref int Total, string Sorts = "CREATED_DATE desc")
        {
            try
            {
                PayrollRepository rep = new PayrollRepository();
                return rep.GetHoldSalaryList(PeriodId, OrgId, IsDissolve, log, PageIndex, PageSize, Total, Sorts);
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }

        public bool InsertHoldSalary(List<PAHoldSalaryDTO> objPeriod, UserLog log)
        {
            try
            {
                PayrollRepository rep = new PayrollRepository();
                return rep.InsertHoldSalary(objPeriod, log);
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }

        public bool DeleteHoldSalary(List<decimal> lstDelete)
        {
            try
            {
                PayrollRepository rep = new PayrollRepository();
                return rep.DeleteHoldSalary(lstDelete);
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }
        #endregion
    }
}
